modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -420,33 +419,46 @@ private void saveGroupConfiguration(AbstractWriteBatch writeBatch, byte[] config
     /**
      * Writes a tuple of transaction id and a row bytes, using "row prefix" of a key array as a storage key.
      *
-     * @param keyArray Array that has partition id and row id in its prefix.
+     * @param directKeyBuffer Buffer that has row id in its prefix.
      * @param row Binary row, not null.
      * @param txId Transaction id.
      * @throws RocksDBException If write failed.
      */
-    private void writeUnversioned(byte[] keyArray, BinaryRow row, UUID txId, int commitTableId, int commitPartitionId)
+    private void writeUnversioned(ByteBuffer directKeyBuffer, BinaryRow row, UUID txId, int commitTableId, int commitPartitionId)
             throws RocksDBException {
         @SuppressWarnings("resource") WriteBatchWithIndex writeBatch = PartitionDataHelper.requireWriteBatch();
 
-        byte[] rowBytes = rowBytes(row);
+        ByteBuffer value = allocateDirect(row.length() + VALUE_HEADER_SIZE);
```

Comment:
Same here. We could use `allocate` + `wrapPointer`, and then free it manually, That would be better I think.